src/sage/matrix/special.py

```diff
@@ -1000,11 +1000,23 @@ def zero_matrix(ring, nrows=None, ncols=None, sparse=False):
         [0 0 0 0 0]
         [0 0 0 0 0]
 
+    TESTS:
+
+    Check that :issue:`38221` is fixed::
+
+        sage: # needs sage.groups
+        sage: G = CyclicPermutationGroup(7)
+        sage: R = GF(2)
+        sage: A = G.algebra(R)
+        sage: zero_matrix(A, 3, 3)
```

Comment:
This is the same test as in `matrix_space.py`. It would be better to have separate tests that better isolate the need for the changes. If they cannot be separated (which is quite possible), then there isn't a point in copying the test.

Reply:
It's not the same test. Here it is tests the global `zero_matrix` function, there it tests the `zero_matrix` method of `MatrixSpace`. Two completely separate code paths.

Reply:
Ah, right, sorry.